### PR TITLE
Add a LiteratureView

### PR DESCRIPTION
**Pull request by brianhelba**

### CODE REVIEW

File: web_external/Literature/LiteratureView.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            if (aYear < bYear) {
+                return 1;
+            } else if (aYear > bYear) {
+                return -1;
+            } else {
+                const aAuthor = a.entryTags.author || '';
+                const bAuthor = b.entryTags.author || '';
+                return aAuthor > bAuthor ? 1 : -1;
+            }
+        });
```
Comment (msmolens) [inline web_external/Literature/LiteratureView.js]: Can you add a short description of the comparator? E.g. "Sort BibTeX entries by year, in descending order, then by author.

File: web_external/Literature/LiteratureView.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            .replace(/\\&/g, '&amp;')
+            .replace(/{\\`([aeiou])}/g, '&$1grave;')
+            .replace(/{\\'([aeiouy])}/g, '&$1acute;')
+            .replace(/{\\^([aeiou])}/g, '&$1circ;')
+            .replace(/{\\~([ano])}/g, '&$1tilde;')
+            .replace(/{\\"([aeiouy])}/g, '&$1uml;')
+            .replace(/{\\c([c])}/g, '&$1cedil;')
+            .replace(/{\\i}/g, '&#x0131;')
+            .replace(/{\\l}/g, '&#x0142;');
+    },
```
Comment (msmolens) [inline web_external/Literature/LiteratureView.js]: Would this package help? https://www.npmjs.com/package/latex-to-unicode-converter

File: web_external/Literature/literaturePage.pug [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                td= entryTags.year
+                case entryType
+                  when 'article'
+                    td Journal
+                  when 'book'
+                    td Book
+                  when 'conference'
+                    td Conference
+                  when 'inbook'
+                    tb Book Section
```
Comment (msmolens) [inline web_external/Literature/literaturePage.pug]: Should that be `td`?

File: web_external/Literature/literaturePage.styl [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        th.isic-literature-column-year
+          width 60px
+        th.isic-literature-column-type
+          width 110px
+    tbody
+      tr
+        &:nth-child(even)
+          background-color #eeeeee
+        &:nth-child(odd)
+          background-color #ffffff
```
Comment (msmolens) [inline web_external/Literature/literaturePage.styl]: Bootstrap has `.table-striped`, but if it doesn't do exactly what you want out of the box you'll probably end up using rules like this anyway.

File: web_external/Literature/literaturePage.pug (hunk 3 above)
Comment (brianhelba) [inline web_external/Literature/literaturePage.pug]: Fixed.

**Comment by kallooa** [comment]
@brianhelba PTAL

**Comment by brianhelba** [comment]
@msmolens PTAL. I've now improved the engineering of the Unicode conversion to be general enough to be future-proof. I've also addressed all your previous review comments.

File: loaders/bibtex-loader.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+const _ = require('underscore');
+const BibtexParse = require('bibtex-parse-js');
+// Use destructiring assignment to import only the "latex2unicode" named export
```
Comment (msmolens) [inline loaders/bibtex-loader.js]: *destructuring

File: loaders/bibtex-loader.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+const { latex2unicode } = require('mathy-unicode-characters');
+
+function unescapeLatexUnicode(str) {
+    // Our BibTex sources (from Google Scholar) use non-standard character escape sequences
```
Comment (msmolens) [inline loaders/bibtex-loader.js]: *BibTeX

**Comment by brianhelba** [comment]
@msmolens Fixed typos, and handled errors when the code lookup failed entirely too.

**Pull request merged**